### ISSUE TITLE
color-contrast incorrectly calculating background color of an element in closed `<details>`

### ISSUE THREAD

**Issue by wojtekmaj** [dequelabs/axe-core#3532]
Product: axe-core

Expectation: When calculating contrast ratio for an element in closed `<details>`, either:
* take background color of open `<details>` element (might not be possible without page interaction, which could be a roadblock), 
* or don't take any background color at all, effectively skipping the color-contrast test for this element.
* or if we really really would like to take background color of where the element _would have been_ if the `<details>` element was open, ignore `<details>`'s siblings when calculating background colors, because they will be pushed down when `<details>` is eventually opened.

Actual: background color is taken from the place where the element _would have been_  if the `<details>` element was open. Unfortunately, in our example, in this place we have a dark blue button which, when combined with black text, fails the test.

Motivation: Current behavior is obviously faulty.

<pre><code>
axe-core version: checked on: 4.0.0, …, 4.1.0, …, 4.2.0, …, 4.3.0, …, 4.4.0, …, 4.4.2

Browser and Assistive Technology versions
Chrome 103.0.5060.53
</code></pre>

Repro:

```html
<html lang="en-US">
  <head>
    <style>
      span {
        position: relative;
      }

      button {
        background-color: darkblue;
        color: #fff;
      }
    </style>
    <title>color-contrast bug</title>
  </head>

  <body>
    <header>
      <h1>color-contrast bug</h1>
    </header>
    <main>
      <details>
        <summary>Summary</summary>
        <span>Some text</span>
      </details>
      <button>A blue button</button>
    </main>
  </body>
</html>

```

**Comment by wojtekmaj** [comment]
It appears that the problem is in `filteredRectStack` function, which in this case, returns `[span, button, main, body, html]`.

**Comment by WilcoFiers** [comment]
Thanks for reporting. We'll take a closer look ASAP.

**Comment by straker** [comment]
Looking at the issue, it appears we don't treat elements inside the `<details>` as hidden when the element is collapsed. As is, we think the `<span>` element is currently visible and positioned right above the button, so that's why we reporting the overlapping element.

I believe `<details>` elements have the `open` attribute when expanded, so we should be able to use that to determine visibility.

**Comment by straker** [comment]
Need to discuss with Wilco the ramifications of saying elements inside closed detail elements are hidden as we may not run axe-core rules against them.

**Comment by straker** [comment]
We discussed this and the resolution is we'll update the is-hidden function to account for this. This means rules will not run against elements inside of a closed <details> elements and users will need to expand the details in order to get accessibility checks.

**Comment by padmavemulapati** [comment]
Validated with the latest develop branch code base, and here  for the test script,  inside `<details>` closed tag <summary> and <span> are passing the `color-contrast` rule as it is counting the elements inside closed details as `hidden`
```
<main>
      <details>
        <summary>Summary</summary>
        <span>Some text</span>
      </details>
      <button>A blue button</button>
    </main>
```

<img width="1420" alt="image" src="https://user-images.githubusercontent.com/47409579/196710612-7c43eb9b-ec51-49a8-81b6-578756b0615d.png">
